src/main/kotlin/dev/roava/group/Group.kt

```diff
@@ -283,4 +282,92 @@ class Group {
     fun exileUser(user: User) {
         exileUser(user.id)
     }
+
+    /**
+     * Method to grab all group members with the provided roleset
+     *
+     * @return[GroupRankListData]
+     * @param[roleSetId] The roleSetId of the group rank.
+     * @throws[RuntimeException]
+     */
+    @Throws(RuntimeException::class)
+    fun getGroupRankMembers(roleSetId: Int): List<GroupRankListData>{
+        val members: MutableList<GroupRankListData> = mutableListOf()
+        var nextPageCursor: String? = null
+        fun makeRequest(){
+            val result = runCatching {
+                request.createRequest(GroupApi::class.java, "groups")
+                    .getGroupRankMembers(this.id,roleSetId, 100, nextPageCursor)
+                    .execute()
+            }
+            result.onFailure { exception ->
+                if (exception is HttpException) {
+                    val errorCode = exception.code()
+                    val message = exception.message()
+
+                    throw RuntimeException("Grabbing members in the group with id ${this.id} & roleSetId of $roleSetId failed with message \"$message\" and response code $errorCode")
+                } else {
+                    throw RuntimeException("an unknown error has occurred while fetching the members with that rank!\n${exception.message}")
+                }
+            }
+            result.onSuccess {
+                nextPageCursor = it.body()?.nextPageCursor
+                val data = it.body()?.data ?: throw RuntimeException("An unknown error has occurred")
+                for(i in data){
+                    members += GroupRankListData(i.hasVerifiedBadge,i.userId,i.username,i.displayName)
+                }
+            }
+        }
```

Comment:
This entire function within this method should be abstracted and more generalized so that we can use it for any paginated request.